### ISSUE TITLE
qx.ui.layout.Canvas and marginLeft seem to affect the right margin.

### ISSUE THREAD

**Issue by lucapivato** [qooxdoo/qooxdoo#10037]
I think it's a bug but I'm not sure.

Try this:
https://bit.ly/3hChX3w

Line 257 in https://github.com/qooxdoo/qooxdoo/blob/master/framework/source/class/qx/ui/layout/Canvas.js

Remove the marginLeft from the left position when the right property is set. It seems wrong that setting the left margin moves a panel that has set a right distance further to the left.

The same issue is present with marginTop. Can someone else verify please?

**Comment by oetiker** [comment]
Indeed, that behavior is distinctly odd ... also when setting both a left margin and a right margin they get added ... 

the problem might be rooted in how it acts if you were to define left and right when adding this to the layout ... not quite clear what would/should happen then.

**Comment by lucapivato** [comment]
I think the behavior should be similar to css with absolute positioning. It seems that:

- When left is specified the right margin is ignored.
- When right is specified, the left margin is ignored.

Which makes sense, the left position trumps the left margin and viceversa. The same is for top and bottom.

I have changed in wisej-qx build since it was breaking some theme appearances and I also added support for alignX and alignY in qx.ui.layout.Canvas.js which was missing, but I don't know if it's a change that is too risky to put in the main.

```
<div style="background-color:red;width:300px;height:300px;position:absolute">
  <div style="background-color:green;position:absolute;right:0px;width:100px;height:100px;margin-left:100px;margin-right:10px">
  </div>
  <div style="background-color:yellow;position:absolute;left:0px;width:100px;height:100px;margin-left:10px;margin-right:1000px">
  </div>
</div>
```

**Comment by oetiker** [comment]
well, make it a PR and then we see what the others have to say about it.

**Comment by hkollmann** [comment]
IMHO changing this will break BC
